ldes-fragmentisers/ldes-fragmentisers-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentation/FragmentationService.java

```diff
@@ -1,35 +1,149 @@
+	private final PlatformTransactionManager transactionManager;
+	private final ItemReader<IngestedMember> newMemberReader;
+	private final ItemReader<IngestedMember> rebucketiseMemberReader;
+	private final BucketProcessor processor;
+	private final ItemWriter<List<BucketisedMember>> bucketWriter;
+	private final ApplicationEventPublisher eventPublisher;
+	private AtomicBoolean shouldTriggerBucketisation = new AtomicBoolean(false);
 
-	private final FragmentationStrategyCollection fragmentationStrategyCollection;
 	private final FragmentRepository fragmentRepository;
 
-	public FragmentationService(FragmentationStrategyCollection fragmentationStrategyCollection,
-								FragmentRepository fragmentRepository) {
-		this.fragmentationStrategyCollection = fragmentationStrategyCollection;
-        this.fragmentRepository = fragmentRepository;
-    }
+	public FragmentationService(JobLauncher jobLauncher, JobExplorer jobExplorer, JobRepository jobRepository,
+	                            PlatformTransactionManager transactionManager,
+	                            @Qualifier("newMemberReader") ItemReader<IngestedMember> newMemberReader,
+	                            @Qualifier("refragmentEventStream") ItemReader<IngestedMember> rebucketiseMemberReader,
+	                            BucketProcessor processor,
+	                            ItemWriter<List<BucketisedMember>> bucketWriter, ApplicationEventPublisher eventPublisher,
+	                            FragmentRepository fragmentRepository) {
+		this.jobLauncher = jobLauncher;
+		this.jobExplorer = jobExplorer;
+		this.jobRepository = jobRepository;
+		this.transactionManager = transactionManager;
+		this.newMemberReader = newMemberReader;
+		this.rebucketiseMemberReader = rebucketiseMemberReader;
+		this.processor = processor;
+		this.bucketWriter = bucketWriter;
+		this.eventPublisher = eventPublisher;
+		this.fragmentRepository = fragmentRepository;
+	}
 
 	@EventListener
 	public void executeFragmentation(MembersIngestedEvent event) {
-		fragmentationStrategyCollection
-				.getFragmentationStrategyExecutors(event.collectionName())
-				.forEach(FragmentationStrategyExecutor::execute);
+		shouldTriggerBucketisation.set(true);
+	}
+
+	@EventListener
+	@Async
+	public void handleViewInitializationEvent(ViewNeedsRebucketisationEvent event) throws JobInstanceAlreadyCompleteException, JobExecutionAlreadyRunningException, JobParametersInvalidException, JobRestartException {
+		launchJob(rebucketiseJob(), new JobParametersBuilder()
+				.addString("viewName", event.viewName().asString())
+				.addLocalDateTime("triggered", LocalDateTime.now())
+				.toJobParameters());
+	}
+
+	@Scheduled(fixedRate = 1500)
+	public void scheduledJobLauncher() throws JobInstanceAlreadyCompleteException, JobExecutionAlreadyRunningException, JobParametersInvalidException, JobRestartException {
+		if (shouldTriggerBucketisation.get() && !isJobRunning(BUCKETISATION_JOB) && !isJobRunning(REBUCKETISATION_JOB)) {
```

Comment:
I might be misunderstanding but if a new view is added to an already large collection and the rebucketisation takes a long time, won't this starve out the normal bucketisation?

Reply:
although correct.. Thing is that if you don't do this, you risk concurrency on a specific bucket